### PR TITLE
modifiers: add request modifier that removes headers by prefix

### LINKED ISSUE #168: util: add request modifier that removes headers by prefix

### PR DESCRIPTION
**Pull request by Choraden**
Fixes #168

### CODE REVIEW

**Comment by Choraden** [comment]
Prefix is now formatted in canonic way. 

I've also done some research on this. 

```go
// Add adds the key, value pair to the header.
// It appends to any existing values associated with key.
func (h MIMEHeader) Add(key, value string) {
	key = CanonicalMIMEHeaderKey(key)
	h[key] = append(h[key], value)
}

// Del deletes the values associated with key.
func (h MIMEHeader) Del(key string) {
	delete(h, CanonicalMIMEHeaderKey(key))
}
```

Methods from http package auto translate keys to canonic way. Using this methods is safe. The only way one can break it is to create header in place: 
```go 
http.Header{
    "non canonical": nil,
},
```
But that would be a really stupid user. 

As long as forwarder is using http.ReadRequest we're good. 

@mmatczuk

**Comment by mmatczuk** [comment]
Package shall be singular. Anyway, let's I'd like to suggest a few changes:
* Put that to http_proxy.go
* Do not export
* Rename to something more flexible ex. headerRemover
* Add this feature to proxy config so that it's usable from CLI

File: bind/flag.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ func HTTPProxyConfig(fs *pflag.FlagSet, cfg *forwarder.HTTPProxyConfig) {
 		"upstream-proxy", "u", "upstream proxy URL")
 	fs.BoolVarP(&cfg.ProxyLocalhost, "proxy-localhost", "t", cfg.ProxyLocalhost,
 		"proxy localhost requests to an upstream proxy")
+	fs.StringSliceVar(&cfg.HeaderRemover, "header-remover", cfg.HeaderRemover, "removes request headers by prefix (can be specified multiple times)")
```
Comment (mmatczuk) [inline bind/flag.go:37]: can we have a sane flag name for that?

Comment (mmatczuk) [reply bind/flag.go]: Shorter plese.

Comment (Choraden) [reply bind/flag.go]: `remove-headers` sounds nice to me

File: http_proxy.go [hunk 2]
```diff
@@ -65,6 +66,7 @@ type HTTPProxyConfig struct {
 	RequestModifiers  []martian.RequestModifier  `json:"-"`
 	ResponseModifiers []martian.ResponseModifier `json:"-"`
 	CloseAfterReply   bool                       `json:"close_after_reply"`
+	HeaderRemover     []string                   `json:"header_remover"`
```
Comment (mmatczuk) [inline http_proxy.go:69]: ?

Comment (Choraden) [reply http_proxy.go]: changed to flag name i.e. `RemoveHeaders`

**Pull request merged**